src/mjolnir/valhalla_convert_transit.cc

```diff
@@ -1,35 +1,89 @@
 #include "baldr/rapidjson_utils.h"
+#include "filesystem.h"
 #include "mjolnir/convert_transit.h"
 #include "mjolnir/validatetransit.h"
 
-int main(int argc, char** argv) {
-  if (argc < 2) {
+#include "config.h"
+#include <cxxopts.hpp>
+
+filesystem::path config_file_path;
+boost::property_tree::ptree pt;
+std::vector<valhalla::mjolnir::OneStopTest> onestoptests;
+
+bool ParseArguments(int argc, char* argv[]) {
+  try {
+    // clang-format off
+    cxxopts::Options options(
+      "valhalla_convert_transit",
+      "valhalla_convert_transit " VALHALLA_VERSION "\n\n"
+      "valhalla_convert_transit is a program that reads protobuf files and creates Level 3 Transit Tiles."
+      "\n\n");
+
+    options.add_options()
+      ("h,help", "Print this help message.")
+      ("v,version", "Print the version of this software.")
+      ("c,config", "Path to the json configuration file.", cxxopts::value<std::string>())
+      ("target_directory", "Path to write transit tiles", cxxopts::value<std::string>())
+      ("test_file", "file where tests are written", cxxopts::value<std::string>());
```

Comment:
in the next part we can change to get target from config and the test file can be a flag rather than positional